osu.Game/Screens/Menu/LogoVisualisation.cs

```diff
@@ -70,6 +70,7 @@ public class LogoVisualisation : Drawable, IHasAccentColour
 
         private IShader shader;
         private readonly Texture texture;
+        public bool IsIntro = false;
```

Comment:
Flags added for specific purposes like this are always asking for trouble. It's doing things backwards; to reuse this visualisation for the legacy intro we should look for ways to *remove* the user colour changing logic for that particular use case entirely, and to keep it in all the other places that need it right now, and not ways to override it.

I can't say I see a very good way of achieving this right now, but I think extracting out a simplified variant of `LogoVisualisation` that doesn't have `User` or `Skin`, and then inheriting from it to add that functionality back where it's needed right now could be a decent idea to start with. That might end up being a separate prerequisite PR, however, depending if any renames/significant code reshuffle happens.

Reply:
I've implemented this in https://github.com/ppy/osu/pull/9236. The naming for the new base class for `LogoVisualisation` might be a bit questionable though.